src/backend/commands/matview.c

```diff
@@ -1501,19 +1565,48 @@ make_delta_enr_name(const char *prefix, Oid relid, int count)
 	return name;
 }
 
+static bool
+checkQueryHasRtable_walker(Query *query, Oid relid)
+{
+	ListCell *lc;
+
+	/* This can recurse, so check for excessive recursion */
+	check_stack_depth();
+
+	foreach(lc, query->rtable)
+	{
+		RangeTblEntry *r = (RangeTblEntry *) lfirst(lc);	
+		if (r->relid == relid)
+			return true;
+		else if (r->subquery)
+		{
+			if (checkQueryHasRtable_walker(r->subquery, relid))
+				return true;
+		}
+	}
+	return false;
+
+}
+
 static void
 register_delta_ENRs(ParseState *pstate, Query *query, List *tables)
 {
 	QueryEnvironment *queryEnv = pstate->p_queryEnv;
 	ListCell *lc;
 	RangeTblEntry	*rte;
+	bool flag = false;
 
 	foreach(lc, tables)
 	{
 		MV_TriggerTable *table = (MV_TriggerTable *) lfirst(lc);
 		ListCell *lc2;
 		int count;
 
+		flag = checkQueryHasRtable_walker(query, table->table_id);
+
+		if (!flag)
+			continue;
+
```

Comment:
I am not sure why this check is necessary, but if we need, we can do just like
```
if(!checkQueryHasRtable_walker(...))
   continue;
```

so I think this flag variable is unnecessary.